CQC dynamic sim

Cyruz143 opened the pull request:
Still needs some performance testing in large scale environments to see if it makes a difference. Holding til tested.

Cyruz143 commented:
Quick preliminary test:

Test mission on Fallujah with 500 CQC units in one zone 3km zone. Player at 0,0,0

DynamicSim off = 60fps
DynamicSim on = 100fps

Obivously this won't be as drastic in proper MP and it's showing a huge change because of the stupid number of units but it looks like there is some performance to be had here. Needs some further proper dedi testing to see if anything else breaks. The forceFire bits could use a `simulationEnabled` check to skip over units to save a bit more performance too. Will continue to test.

Cyruz143 commented:
Did a bit more testing tonight. The only negative I can find is that if you shoot a unit that's currently simulation disabled, it won't die. However when it does "turn on" due to getting close enough, it collapses and dies. I've set it at 500m by default with a 1.5x modifier when moving (so 750m when driving). I don't think this would be a massive deal with our play style (no scopes/snipers etc), only time I could see it being an issue is with vehicle coax. Would like thoughts on this.

Cyruz143 commented on admiral/cqc_functions.sqf:
```diff
@@ -116,7 +116,6 @@ adm_cqc_fnc_spawnGarrisonGroupUnits = {
         private _position = selectRandom _possiblePositions;
         _possiblePositions = _possiblePositions - [_position];
         private _unit = [_building buildingPos _position, _group, _unitTemplate, _zoneTemplate, UNIT_TYPE_ARRAY select UNIT_TYPE_INF] call adm_cqc_fnc_placeMan;
-        _unit setVariable ["adm_unit_isCqc", true, true];
```
I added this, it used to be used in our custom danger.fsm (which we binned because FSMs are slooow).

I don't use it in inhouse anywhere and it's not referenced in hull or admiral anywhere, can reinstate if required but I was trying to cut down on unused global variables.

Cyruz143 replied:
Keep or remove @kami- ?

If keep can we at least make it a local variable as all AI is generally server owned so clients don't need access to the variable.

kami- replied:
Sorry, you can remove it.

Cyruz143 commented:
Adjusted this to make it much cleaner.

Removed the setup function as `enableDynamicSimulationSystem` is true by default and you can adjust the values via Eden (less config if the mission maker wants to actually adjust something). The default values are fine too.
